### PR TITLE
Only warn on links to bsky.app if it represents itself as another url

### LINKED ISSUE #1652: Fix/remove the interstitial for internal links
**Describe the bug**
The interstitial added by #1573 warns that we leave Bluesky even if we actually don't, when making a link to bsky.app with a regular text (not a 'false URL')

**To Reproduce**

1. Create a link to an element of bsky.app (post, feed...) but with a regular text (like : "check my thread here")
2. Click on the link

**Expected behavior**

Either change the text of the card link, or remove the card link? I don't know what would be best.

**Screenshots**

image
image

**Details**

 - Platform: Firefox 
 - Platform version: 118.0.1 Windows
 - App version: 1.5.2 I think?

**Additional context**

This feature could also be used to create abuse (like writing insults that lead to someone's account). That might be taken into consideration.

### PR DESCRIPTION
**Pull request by pfrazee**
Closes #1652

Updates the link-warning interstitial to only trigger for `bsky.app` links if the richtext represents itself as another URL. 

CleanShot 2023-10-10 at 12 26 10@2x
CleanShot 2023-10-10 at 12 26 18@2x

### CODE REVIEW

File: src/state/models/root-store.ts [hunk 1]
```diff
@@ -223,6 +223,7 @@ export class RootStoreModel {
   }
   emitSessionReady() {
     DeviceEventEmitter.emit('session-ready')
+    window.agent = this.agent
```
Comment (gaearon) [inline src/state/models/root-store.ts:226]: ?

File: src/lib/strings/url-helpers.ts [hunk 2]
```diff
@@ -170,15 +170,36 @@ export function getYoutubeVideoId(link: string): string | undefined {
...
-    const urip = new URL(uri)
-    return labelDomain !== urip.hostname
+    urip = new URL(uri)
   } catch {
     return true
   }
+
+  if (urip.hostname === 'bsky.app') {
+    // if this is a link to internal content,
+    // warn if it represents itself as a URL to another app
```
Comment (gaearon) [inline src/lib/strings/url-helpers.ts:182]: what if it's a link to bsky but a different page than claimed?

File: src/lib/strings/url-helpers.ts [hunk 3]
```diff
@@ -170,15 +170,36 @@ export function getYoutubeVideoId(link: string): string | undefined {
...
+    }
+    return false
+  } else {
+    // if this is a link to external content,
+    // warn if the label doesnt match the target
+    if (!labelDomain) {
+      return true
+    }
+    try {
+      return labelDomain !== urip.hostname
```
Comment (gaearon) [inline src/lib/strings/url-helpers.ts:198]: do you expect `.hostname` accessor to throw? why try/catch?

File: src/state/models/root-store.ts (hunk 1 above)
Comment (pfrazee) [reply src/state/models/root-store.ts:226]: of course I forgot to remove that

File: src/lib/strings/url-helpers.ts (hunk 3 above)
Comment (pfrazee) [reply src/lib/strings/url-helpers.ts:198]: eugh moving too quickly. leftover code, will fix that

File: src/lib/strings/url-helpers.ts (hunk 2 above)
Comment (pfrazee) [reply src/lib/strings/url-helpers.ts:182]: this is the meat of it here. if you look at the issue that triggered this, people do thinks like "see [this thread]". the question is what exactly do we want to mitigate, and how

a likely issue would be faking mentions, like doing "@bob.xyz" and it actually goes to "@bob.site". right now that's not getting covered. after that it's just misleading labels like "this post by paul frazee" and it's a post by "@pfrazee.io." I'd rather not do an interstitial on *every* link to internal content that's not a raw URL, so I'm not sure what kinds of mitigations to aim for there

**Review by gaearon — APPROVED** [review]

**Pull request merged**